main.go

```diff
@@ -1,105 +1,54 @@
 	"net/http"
+	"time"
 
 	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-type Table struct {
-	Id          string `json:"Id"`
-	MaxPosition int    `json:"maxPosition"`
-}
-
-type Reserve struct {
-	Id          string `json:"Id"`
-	DateTime    string `json:"maxPosition"`
-	TotalPeople int    `json:"totalPeople"`
-	PhoneNumber string `json:"phoneNumber"`
-	TableId     string `json:"tableId"`
-}
-
-var Reserves = []*Reserve{
-	{Id: "1", DateTime: "21/10/2022", TotalPeople: 2, PhoneNumber: "0799004177", TableId: "1"},
-	{Id: "2", DateTime: "22/10/2022", TotalPeople: 5, PhoneNumber: "0913616957", TableId: "2"},
-}
-
-// Tables := []Table{
-// 	Table{Id: "1", maxPosition: 4},
-// 	Table{Id: "2", maxPosition: 8},
-// }
-
-func homePage(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "Welcome to the HomePage!")
-	fmt.Println("Endpoint Hit: homePage")
-}
-
 func handleRequests() {
 	myRouter := mux.NewRouter().StrictSlash(true)
-	// http.HandleFunc("/", homePage)
-	myRouter.HandleFunc("/", homePage)
-	myRouter.HandleFunc("/booking", createNewBooking).Methods("POST")
-	myRouter.HandleFunc("/getAllTable", getAllTable).Queries("date", "{date}")
-	myRouter.HandleFunc("/cancelBooking", cancelBooking).Queries("phoneNumber", "{phoneNumber}")
-	myRouter.HandleFunc("/updateBooking", updateBooking).Methods("POST")
-	// myRouter.HandleFunc("/all", returnAllArticles)
-	// finally, instead of passing in nil, we want
-	// to pass in our newly created router as the second
-	// argument
-	log.Fatal(http.ListenAndServe(":10000", myRouter))
-}
+	serverAPIOptions := options.ServerAPI(options.ServerAPIVersion1)
+	uri := "mongodb+srv://lieuvinhthu:Abc123456@cluster0.m2rac.mongodb.net/?retryWrites=true&w=majority"
 
-func createNewBooking(w http.ResponseWriter, r *http.Request) {
-	reqBody, _ := ioutil.ReadAll(r.Body)
-	var reserve Reserve
-	json.Unmarshal(reqBody, &reserve)
-	Reserves = append(Reserves, &reserve)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
 
-	json.NewEncoder(w).Encode(Reserves)
-}
-
-func getAllTable(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-	reqDate := vars["date"]
-	var filterReserves []*Reserve
-	for _, reserve := range Reserves {
-		if reserve.DateTime == reqDate {
-			filterReserves = append(filterReserves, reserve)
-		}
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri).SetServerAPIOptions(serverAPIOptions))
+	if err != nil {
+		panic(err)
 	}
-	json.NewEncoder(w).Encode(filterReserves)
-}
 
-func cancelBooking(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-	reqPhoneNumber := vars["phoneNumber"]
-	var filterReserves []*Reserve
-	for _, reserve := range Reserves {
-		if reserve.PhoneNumber != reqPhoneNumber {
-			filterReserves = append(filterReserves, reserve)
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
```

Comment:
try to not use panic that can be side effect. Maybe you should return error directly.